src/Libraries/Microsoft.Extensions.Diagnostics.ResourceMonitoring/Linux/Network/LinuxNetworkUtilizationParser.cs

```diff
@@ -0,0 +1,154 @@
+        // therefore we request 5 ranges instead of 4, and then range[Target - 2] will have the range we need without the remaining elements.
+        int numRanges = line.Split(range, ' ', StringSplitOptions.RemoveEmptyEntries);
+#else
+        const int Target = 4;
+        Span<StringRange> range = stackalloc StringRange[Target];
+
+        // in our StringRange API, if capacity of destination range array is less than number of ranges found by ReadOnlySpan<T>.TrySplit(),
+        // the last range in the array will get the last range as expected, and all remaining elements will be ignored.
+        // hence range[Target - 1] will have the last range as we need.
+        _ = line.TrySplit(" ", range, out int numRanges, StringComparison.OrdinalIgnoreCase, StringSplitOptions.RemoveEmptyEntries);
+#endif
+        if (numRanges < Target)
+        {
+            Throw.InvalidOperationException($"Could not split contents. We expected every line to contain more than {Target - 1} elements, but it has only {numRanges} elements.");
+        }
+
+#if NET8_0_OR_GREATER
+        ReadOnlySpan<char> tcpConnectionState = line.Slice(range[Target - 2].Start.Value, range[Target - 2].End.Value - range[Target - 2].Start.Value);
+#else
+        ReadOnlySpan<char> tcpConnectionState = line.Slice(range[Target - 1].Index, range[Target - 1].Count);
+#endif
+
+        // until this API proposal is implemented https://github.com/dotnet/runtime/issues/61397
+        // we have to allocate & throw away memory using ToString():
+        switch ((LinuxTcpState)Convert.ToInt32(tcpConnectionState.ToString(), Base16))
+        {
+            case LinuxTcpState.ESTABLISHED:
+                tcpStateInfo.EstabCount++;
+                break;
+            case LinuxTcpState.SYN_SENT:
+                tcpStateInfo.SynSentCount++;
+                break;
+            case LinuxTcpState.SYN_RECV:
+                tcpStateInfo.SynRcvdCount++;
+                break;
+            case LinuxTcpState.FIN_WAIT1:
+                tcpStateInfo.FinWait1Count++;
+                break;
+            case LinuxTcpState.FIN_WAIT2:
+                tcpStateInfo.FinWait2Count++;
+                break;
+            case LinuxTcpState.TIME_WAIT:
+                tcpStateInfo.TimeWaitCount++;
+                break;
+            case LinuxTcpState.CLOSE:
+                tcpStateInfo.ClosedCount++;
+                break;
+            case LinuxTcpState.CLOSE_WAIT:
+                tcpStateInfo.CloseWaitCount++;
+                break;
+            case LinuxTcpState.LAST_ACK:
+                tcpStateInfo.LastAckCount++;
+                break;
+            case LinuxTcpState.LISTEN:
+                tcpStateInfo.ListenCount++;
+                break;
+            case LinuxTcpState.CLOSING:
+                tcpStateInfo.ClosingCount++;
+                break;
+            default:
+                throw new InvalidEnumArgumentException($"Cannot find status: {tcpConnectionState} in LinuxTcpState");
```

Comment:
❗ Throwing exception like this has perf implications (e.g., methods can't be inlined). Please use [`Throw` helpers](https://github.com/dotnet/extensions/blob/main/src/Shared/Throw/Throw.cs) instead.

Reply:
Done